src/lib.rs

```diff
@@ -311,263 +333,246 @@ impl<'a, Writer: Write> Printer<'a, Writer> {
         }
+        Ok(())
     }
 
-    fn print_position_panel(&mut self) {
-        if !self.show_position_panel {
-            write!(&mut self.buffer_line, "{} ", self.border_style.outer_sep()).ok();
-            return;
-        }
-
-        let style = COLOR_OFFSET.normal();
-        let byte_index = format!("{:08x}", self.idx - 1 + self.display_offset);
-        let formatted_string = if self.show_color {
-            format!("{}", style.paint(byte_index))
-        } else {
-            byte_index
-        };
-        let _ = write!(
-            &mut self.buffer_line,
-            "{}{}{} ",
-            self.border_style.outer_sep(),
-            formatted_string,
-            self.border_style.outer_sep()
-        );
-    }
-
-    pub fn print_char_panel(&mut self) {
-        if !self.show_char_panel {
-            // just write newline if character panel is hidden
-            writeln!(&mut self.buffer_line).ok();
-            return;
-        }
-
-        let len = self.raw_line.len();
-
-        let mut idx = 1;
-        for &b in self.raw_line.iter() {
-            let _ = write!(
-                &mut self.buffer_line,
-                "{}",
-                self.byte_char_panel[b as usize]
-            );
-
-            if idx % 8 == 0 && idx % (u64::from(self.panels) * 8) != 0 {
-                let _ = write!(&mut self.buffer_line, "{}", self.border_style.inner_sep());
-            }
-
-            idx += 1;
-        }
-
-        if len < usize::from(8 * self.panels) {
-            let _ = write!(&mut self.buffer_line, "{0:1$}", "", 8 - len % 8);
-            for _ in 0..(usize::from(8 * self.panels) - (len + (8 - len % 8))) / 8 {
-                let _ = write!(
-                    &mut self.buffer_line,
-                    "{2}{0:1$}",
-                    "",
-                    8,
-                    self.border_style.inner_sep()
-                );
+    fn print_position_panel(&mut self) -> io::Result<()> {
+        self.writer.write_all(
+            self.border_style
+                .outer_sep()
+                .encode_utf8(&mut [0; 4])
+                .as_bytes(),
+        )?;
+        if self.show_position_panel {
+            match self.squeezer {
+                Squeezer::Print => {
+                    self.writer
+                        .write_all(self.byte_char_panel_g[b'*' as usize].as_bytes())?;
+                    self.writer.write_all(b"       ")?;
+                }
+                Squeezer::Ignore | Squeezer::Disabled | Squeezer::Delete => {
+                    let byte_index: [u8; 8] = (self.idx + self.display_offset).to_be_bytes();
+                    let mut i = 0;
+                    while byte_index[i] == 0x0 && i < 4 {
+                        i += 1;
+                    }
+                    for &byte in byte_index.iter().skip(i) {
+                        self.writer
+                            .write_all(self.byte_hex_panel_g[byte as usize].as_bytes())?;
+                    }
```

Comment:
I'm not sure I understand what's going on here.

Reply:
I totally get it, this part is a little trick I made.

Basically, the `byte_index` is the position turned into a big-endian byte array. For example, the position `0x1234` would become `[0x00, 0x00, 0x00, 0x00, 0x12, 0x34, 0x56, 0x78]`. The `while` loop will increment `i` until it finds the first byte in the sequence that is not 0. It will always print out the last four bytes regardless, which is why the loop will break if `i` is 4.

The `for` loop will print out bytes in the array after skipping `i` bytes.